packages/compiler/lib/decorators.ts

```diff
@@ -419,6 +432,85 @@ export function getMaxValue(program: Program, target: Type): number | undefined
   return program.stateMap(maxValuesKey).get(target);
 }
 
+// -- @minValueExclusive decorator ---------------------
+
+const minValueExclusiveKey = createStateSymbol("minValueExslusive");
+
+export function $minValueExclusive(
+  context: DecoratorContext,
+  target: Scalar | ModelProperty,
+  minValueExclusive: number
+) {
+  validateDecoratorUniqueOnNode(context, target, $minValueExclusive);
+  validateDecoratorNotOnNode(context, target, $minValue, $minValueExclusive);
+  const { program } = context;
+
+  if (!isNumericType(program, getPropertyType(target))) {
+    program.reportDiagnostic(
+      createDiagnostic({
+        code: "decorator-wrong-target",
+        format: { decorator: "@minValueExclusive", to: "non-numeric type" },
+        target,
+      })
+    );
+    return;
+  }
+
+  if (
+    !validateRange(
+      context,
+      minValueExclusive,
+      getMaxValue(context.program, target) ?? getMaxValueExclusive(context.program, target)
+    )
+  ) {
+    return;
+  }
+  program.stateMap(minValueExclusiveKey).set(target, minValueExclusive);
+}
+
+export function getMinValueExclusive(program: Program, target: Type): number | undefined {
+  return program.stateMap(minValueExclusiveKey).get(target);
+}
+
+// -- @maxValue decorator ---------------------
+
+const maxValueExclusiveKey = createStateSymbol("maxValueExclusive");
+
+export function $maxValueExclusive(
+  context: DecoratorContext,
+  target: Scalar | ModelProperty,
+  maxValueExclusive: number
+) {
+  validateDecoratorUniqueOnNode(context, target, $maxValueExclusive);
+  validateDecoratorNotOnNode(context, target, $maxValue, $maxValueExclusive);
+  const { program } = context;
+  if (!isNumericType(program, getPropertyType(target))) {
+    program.reportDiagnostic(
+      createDiagnostic({
+        code: "decorator-wrong-target",
+        format: { decorator: "@maxValue", to: "non-numeric type" },
+        target,
+      })
+    );
+    return;
+  }
+
+  if (
+    !validateRange(
+      context,
+      getMinValue(context.program, target) ?? getMinValueExclusive(context.program, target),
+      maxValueExclusive
+    )
+  ) {
+    return;
+  }
+  program.stateMap(maxValueExclusiveKey).set(target, maxValueExclusive);
+}
+
+export function getMaxValueExclusive(program: Program, target: Type): number | undefined {
```

Comment:
Do we need new tests for those decorators?

Reply:
Yep just added them.